tests/test_linear_feedback_controller.cpp

```diff
@@ -0,0 +1,378 @@
+  std::make_tuple(std::string_view{#val}, std::ref((val)))
+
+namespace {
+
+/**
+ *  @return A predicate used to compare 2 double Eigen::Matrix returning true
+ *          when all elements follows |lhs - rhs| <= eps
+ *
+ *  @param[in] abs_error The max absolute error allowed when comparing doubles
+ */
+constexpr auto AreAlmostEquals(double abs_error) {
+  return [=](const auto& lhs, const auto& rhs) -> bool {
+    return ((lhs.array() - rhs.array()).abs() <= abs_error).all();
+  };
+}
+
+struct PDParams {
+  Gains gains;
+  References refs;
+
+  static auto From(const ControllerParameters& params) -> PDParams {
+    constexpr auto ToEigen = [](const std::vector<double>& v) {
+      return Eigen::Map<const Eigen::VectorXd>(v.data(), v.size());
+    };
+
+    PDParams out;
+    out.gains.p = ToEigen(params.p_gains);
+    out.gains.d = ToEigen(params.d_gains);
+    out.refs = References::Random(params.d_gains.size());
+    return out;
+  }
+};
+
+struct Timestamps {
+  TimePoint first_call;
+  TimePoint pd_timeout;
+
+  static auto From(const ControllerParameters& params) -> Timestamps {
+    Timestamps out;
+    out.first_call = TimePoint::clock::now();
+    out.pd_timeout = out.first_call + params.pd_to_lf_transition_duration;
+    return out;
+  }
+};
+
+struct ControllerInputs {
+  Sensor sensor;
+  Control control;
+
+  static auto From(const RobotModelBuilder& model) -> ControllerInputs {
+    ControllerInputs out;
+    out.sensor = MakeValidRandomSensorFor(model);
+    out.control = MakeValidRandomControlFor(model);
+    return out;
+  }
+};
+
+struct Expectations {
+  Eigen::VectorXd pd;
+  Eigen::VectorXd lf;
+
+  static auto From(const RobotModelBuilder& model, const Gains& gains,
+                   const References& refs, const Sensor& sensor,
+                   const Control& control) -> Expectations {
+    Expectations out;
+    out.pd = ExpectedPDControlFrom(gains, refs, sensor.joint_state.position,
+                                   sensor.joint_state.velocity);
+    out.lf = ExpectedLFControlFrom(model, sensor, control);
+    return out;
+  }
+};
+
+struct LinearFeedbackControllerTest
+    : public ::testing::TestWithParam<ControllerParameters> {};
+
+TEST(LinearFeedbackControllerTest, Ctor) {
+  EXPECT_NO_THROW({ auto ctrl = LinearFeedbackController{}; });
+}
+
+TEST(LinearFeedbackControllerTest, DISABLED_LoadEmptyParams) {
+  auto ctrl = LinearFeedbackController{};
+  EXPECT_FALSE(ctrl.load(ControllerParameters{}));
+  EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+}
+
+TEST_P(LinearFeedbackControllerTest, DISABLED_LoadNoURDF) {
+  auto ctrl = LinearFeedbackController{};
+  auto no_urdf_param = GetParam();
+  no_urdf_param.urdf.clear();
+
+  EXPECT_FALSE(ctrl.load(no_urdf_param));
+  EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+}
+
+TEST_P(LinearFeedbackControllerTest, DISABLED_LoadSizeMismatch) {
+  auto ctrl = LinearFeedbackController{};
+  auto good_params = GetParam();
+
+  {
+    auto p_gains_too_big = good_params;
```

Comment:
Maybe a dumb complaint, but it took me a moment to realize "too_big" relates to vector size and not to the value. I know "gains" means plural, but maybe a bit more verbose name? Like `p_gains_vector_too_big`?